frame/clearing-house/src/math.rs

```diff
@@ -26,6 +26,22 @@ where
 //                                             Traits
 // -------------------------------------------------------------------------------------------------
 
+pub trait TryClamp: Ord + Sized {
+	fn try_clamp(self, min: Self, max: Self) -> Result<Self, &'static str> {
+		if min > max {
+			return Err("min must be less than or equal to max")
+		}
+
+		Ok(if self < min {
+			min
+		} else if self > max {
+			max
+		} else {
+			self
+		})
+	}
+}
+
```

Comment:
Nice addition to `math.rs`!

Reply:
What if I told most of it was GitHub copilot?

Reply:
i recall it can be simolified via .max  . bot sure so if need error here at all. why not just use .max directly?

Reply:
The alternative was to use `.clamp`, but it panics if `min > max`, instead of giving an error. This implementation is just to make sure we don't shoot ourselves in the foot in the future.